packages/clay-charts/src/ChartBase.js

```diff
@@ -450,6 +452,18 @@ const ChartBase = {
 		}
```

Comment:
There's something fishy here and in `_handleDataChanged`. It looks like this is being triggered when data changes because down there we do `this.columns = data`. This also means that we are not syncing data if this gets set via `columns` instead

We also risk getting in an infinite loop by triggering dataChanged and columnsChanged over and over, so I think the underlying model needs to be better thought out. 

Could you take a closer look?

Reply:
Yes, that's actually something I wanted to check with you. Thanks